File: frontend/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";
import { Product } from "./product";

export class Purchase {

    customer!:Customer;
    shippingAddress!:Address;
    billingAddress!:Address;
    order!:Order;
    orderItems!:OrderItem[];
}

File: backend/spring-boot-ecommerce/src/main/java/com/ecommerce/springbootecommerce/dto/Purchase.java
package com.ecommerce.springbootecommerce.dto;

import com.ecommerce.springbootecommerce.Entity.*;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;

    private Set<OrderItem> orderItems;

}
